aggregator/apps/api/v1/api_client.py

```diff
@@ -92,11 +107,86 @@ def get_data_for_address(self, slug):
 
-        wcivf_response = asyncio.ensure_future(fetch(wcivf_url))
-        self.loop.run_until_complete(wcivf_response)
-        wcivf_result = wcivf_response.result()
-        if wcivf_result["status"] >= 400:
-            raise ApiError(wcivf_result["status"], wcivf_result["json"]["detail"])
+        wcivf_result = proxy_single_request(self.loop, wcivf_url, self.wcivf_params)
         wcivf_json = wcivf_result["json"]
 
         return wdiv_json, wcivf_json
+
+
+class EEApiClient(AsyncApiClient):
+    def clean_query_params(self, params):
+        """
+        Take a django QueryDict object and return another QueryDict
+        containing only the specified keys.
+
+        We should use this both:
+        - before passing a query string from devs.DC to EE and
+        - before passing a URL with a query string from EE
+          back to the devs.DC user.
+
+        There are several things this acheives:
+        1. There are some params that the devs.DC API accepts
+           that we don't want to pass through to the EE API (e.g: auth_token)
+        2. There are some params supported by the EE API
+           that we don't want to expose to dev.DC users (e.g: postcode)
+        3. In future, if we wanted to pass additional params to EE
+           (e.g: a tracking param or API key, like we do with WDIV),
+           they'll come back reflected in the next/previous URLs.
+           Cleaning prevents us exposing those params to the devs.DC user.
+        """
+        allowed_query_params = [
+            "limit",
+            "offset",
+            "current",
+            "future",
+            "coords",
+            "metadata",
+        ]
+        q = QueryDict("", mutable=True)
+        q.update({k: v for k, v in params.items() if k in allowed_query_params})
+        return q
+
+    def get_single_election(self, slug):
+        ee_url = f"{settings.EE_BASE_URL}elections/{slug}/"
+        ee_result = proxy_single_request(self.loop, ee_url)
+        return self.filter_single_election(ee_result["json"])
+
+    def get_election_list(self, query_dict):
+        ee_url = f"{settings.EE_BASE_URL}elections/"
+        ee_result = proxy_single_request(
+            self.loop, ee_url, dict(self.clean_query_params(query_dict).items())
+        )
+        return self.filter_election_list(ee_result["json"])
+
+    def filter_single_election(self, election):
+        election.pop("deleted", None)
+        election.pop("explanation", None)
+        election.pop("tmp_election_id", None)
+
+        if "organisation" in election and election["organisation"]:
+            election["organisation"].pop("url", None)
+
+        if "division" in election and election["division"]:
+            election["division"].pop("geography_curie", None)
+            election["division"].pop("mapit_generation_low", None)
+            election["division"].pop("mapit_generation_high", None)
+            election["division"]["divisionset"].pop("mapit_generation_id", None)
```

Comment:
There are some of these things I actually want to completely remove in EE, but doing so may impact on YNR/WCIVF so for the moment I'm just going to strip them here. I'm using `.pop("field_name", None)` here because it means that we will remove `"field_name"` from the dict if it exists but it won't throw an exception if it doesn't. This makes it a bit easier to co-ordinate because it means I can make those changes in EE without breaking our code here and then tidy up here later.

Reply:
Agree with the approach – FWIW, YNR/WCIVF don't use this anywhere, so it's safe to remove from EE now I reckon.

Reply:
There is one change I need to make, which is switching YNR to use `official_identifier` instead of `geography_curie`:

https://github.com/DemocracyClub/yournextrepresentative/blob/69caa54c3d5326860852298a892048b51b0aa7ac/ynr/apps/elections/uk/every_election.py#L137-L141

`geography_curie` is just an alias for `official_identifier` now though, so there's not really any point exposing both of them in the API - its just confusing to have the same value with 2 different field names.

https://github.com/DemocracyClub/EveryElection/blob/e936ef13168e3a7fb7ac283e4c006e45ba8e43f2/every_election/apps/organisations/models/divisions.py#L115-L117

Reply:
Ah, sorry, I was talking about the mapit stuff 👍